crates/storage_models/src/dispute.rs

```diff
@@ -1 +1,102 @@
+use common_utils::custom_serde;
+use diesel::{AsChangeset, Identifiable, Insertable, Queryable};
+use serde::{Deserialize, Serialize};
+use time::PrimitiveDateTime;
 
+use crate::{enums as storage_enums, schema::dispute};
+
+#[derive(Clone, Debug, Deserialize, Insertable, Serialize, router_derive::DebugAsDisplay)]
+#[diesel(table_name = dispute)]
+#[serde(deny_unknown_fields)]
+pub struct DisputeNew {
+    pub dispute_id: String,
+    pub amount: String,
+    pub currency: String,
+    pub dispute_stage: storage_enums::DisputeStage,
+    pub dispute_status: storage_enums::DisputeStatus,
+    pub payment_id: String,
+    pub merchant_id: String,
+    pub connector_status: String,
+    pub connector_dispute_id: String,
+    pub connector_reason: Option<String>,
+    pub connector_reason_code: Option<String>,
+    pub challenge_required_by: Option<String>,
+    pub dispute_created_at: Option<String>,
+    pub updated_at: Option<String>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Identifiable, Queryable)]
+#[diesel(table_name = dispute)]
+pub struct Dispute {
+    #[serde(skip_serializing)]
+    pub id: i32,
+    pub dispute_id: String,
+    pub amount: String,
+    pub currency: String,
+    pub dispute_stage: storage_enums::DisputeStage,
+    pub dispute_status: storage_enums::DisputeStatus,
+    pub payment_id: String,
```

Comment:
How the dispute is linked to our storage model, the relation-ship model of dispute.

Reply:
A payment_attempt can have multiple disputes. So it's 1-N (1 to many) relationship for payment_attempt with dispute.